### PR TITLE
Makes annotated_mnist use Optax's xent loss.

### PR DESCRIPTION
**Pull request by andsteing**
As mentioned in #2051, it is a bit confusing that we return `nn.log_softmax(x)` from the model and then define the loss as `-jnp.mean(jnp.sum(one_hot_labels * logits, axis=-1))`

This PR updates the [Annotated MNIST](https://colab.sandbox.google.com/github/google/flax/blob/main/docs/notebooks/annotated_mnist.ipynb) to use the same `optax.softmax_cross_entropy()` for loss computation that we already use in [`examples/minst/`](https://github.com/google/flax/tree/main/examples/mnist)

Note that #2051 also has some discussions about a `vmap()` based implementation and performance comparisons, but the current implementation seems to perform better on TPU and similarly on GPU.

### CODE REVIEW

File: docs/notebooks/annotated_mnist.ipynb [hunk 1]
```diff
@@ -1,598 +1,655 @@
...
+        "[MNIST Example](https://github.com/google/flax/tree/main/examples/mnist).\n",
+        "If you see any changes between the two feel free to create a\n",
+        "[pull request](https://github.com/google/flax/compare)\n",
+        "to synchronize this Colab with the actual example."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Check GPU\n",
```
Comment (marcvanzee) [inline docs/notebooks/annotated_mnist.ipynb:25]: This cell seems to be new, why did you add it?

**Review by marcvanzee — APPROVED** [review]

File: docs/notebooks/annotated_mnist.ipynb (hunk 1 above)
Comment (andsteing) [reply docs/notebooks/annotated_mnist.ipynb:25]: To make clear that the example runs on a Colab with GPU.

But looking at the RTD this is actually quite visible now ...
https://flax--2071.org.readthedocs.build/en/2071/notebooks/annotated_mnist.html

What would you prefer? Removing it or changing to `!nvidia-smi -L` that will have a sinle line output?

Comment (marcvanzee) [reply docs/notebooks/annotated_mnist.ipynb:25]: Makes sense, I actually think it is quite nice it shows the GPU info, so I'm fine keeping it as is.

**Review by marcvanzee — APPROVED** [review]

**Review by ayaka14732 — APPROVED** [review]
